### PR TITLE
New Presence Shiro

### LINKED ISSUE #3297: Shiro | Shiro.is
**Prerequisites and essential questions** 
- [ ] Is it a popular site?
- [x] Is the website older than 2 months? 
- [ ] Is the site locked to a specific country/region?
- [ ] Is the site a paid service? (e.g. Netflix, Hulu)
- [ ] Does the website feature NSFW content? (e.g. porn, etc...)
- [ ] Are you a donator/patron?
- [x] Do you acknowledge that coding presences is completely voluntary and may take time for your service to be added regardless of priority?

**What's your Discord username?** 

**What's the name of the service?** https://shiro.is/

**What should the Presence display?**  Borwsing shiro/watching xxx on shiro

**If possible, please provide a logo for the service (512x512 minimum)** 

On shiro.is you wan watch anime adfree with HD quality. Shiro v.2 will soon be released which will make it one of the best anime sites. currently it does not have a lot of users (2000 accounts) but i am sure it will become a big site once v.2 is released

### PR DESCRIPTION
**Pull request by SlowLife1661**
## Default
image

## Browsing
image

## Seaching
image
image

## Watching anime
image
image

## Viewing anime
image

## Viewing schedule
image
image

- Resolves #3297

### CODE REVIEW

File: websites/S/Shiro/presence.ts [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+
+function YouCanSeeThis(element: HTMLElement) {
+  const clientRect = element.getBoundingClientRect();
+  return (
+    clientRect.top >= 0 &&
+    clientRect.left >= 0 &&
+    clientRect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
+    clientRect.right <= (window.innerWidth || document.documentElement.clientWidth)
+  );
+}
```
Comment (ririxidev) [inline websites/S/Shiro/presence.ts:286]: For what purpose, fr
```typescript
function isHidden(element: HTMLElement) {
    return (!element.offsetParent)
}
``` 
is easier and it has less code.

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @SlowLife1661!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

File: websites/S/Shiro/presence.ts (hunk 1 above)
Comment (SlowLife1661) [reply websites/S/Shiro/presence.ts:286]: That isn't what I want, what I want is just to check if that Element if visble on sceen or not

Comment (ririxidev) [reply websites/S/Shiro/presence.ts:280]: Literally, unnecessary for me but FINE.

File: websites/S/Shiro/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            "id": "AnimeState",
+            "title": "Anime Details row2",
+            "icon": "fas fa-pen-square",
+            "value": "Episode %episode%",
+            "placeholder": "Use %title%, %episode%"
+        },
+        {
+            "id": "ViewAnime",
+            "title": "Viewing Anime row1",
```
Comment (ririxidev) [inline websites/S/Shiro/dist/metadata.json:36]: View

File: websites/S/Shiro/dist/metadata.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+            "icon": "fas fa-pen-square",
+            "value": "Viewing anime:"
+        },
+        {
+            "id": "ViewAnime2",
+            "title": "Viewing Anime row2",
```
Comment (ririxidev) [inline websites/S/Shiro/dist/metadata.json:42]: View

File: websites/S/Shiro/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            "icon": "fas fa-pen-square",
+            "value": "%title%"
+        },
+        {
+            "id": "ViewSchedule",
+            "title": "Viewing Schedule row2",
```
Comment (ririxidev) [inline websites/S/Shiro/dist/metadata.json:48]: View

File: websites/S/Shiro/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            "icon": "fas fa-pen-square",
+            "value": "Viewing schedule of:"
+        },
+        {
+            "id": "ViewSchedule2",
+            "title": "Viewing Schedule row2",
```
Comment (ririxidev) [inline websites/S/Shiro/dist/metadata.json:54]: View

**Review by ririxidev — CHANGES_REQUESTED** [review]
Heya @Slowftw!
You should check out thing(s) that I requested to change in my review. Please look at them!

Thanks,
**ririxi**. PreMiD Reviewer Team.
Contact: mail@ririxi.dev, ririxi#2721

File: websites/S/Shiro/presence.ts (hunk 1 above)
Comment (SlowLife1661) [reply websites/S/Shiro/presence.ts:280]: ;-;

**Review by ririxidev — DISMISSED** [review]
lgtm

File: websites/S/Shiro/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "author": {
+        "name": "SlowLife",
+        "id": "374905512661221377"
+    },
+    "version": "1.0.0",
+    "service": "Shiro",
+    "description": {
+        "en": "If you’re looking for the latest Anime videos of various genres, Shiro is exactly what you need. Here you will find a great collection of Anime videos to stream without ads. Browse our catalog to find the most exciting and popular anime. We have both old-school videos for real admirers and the latest Anime episodes for those who would like to stay up to date."
+    },
```
Comment (Bas950) [inline websites/S/Shiro/dist/metadata.json:11]: please make the description a bit shorter

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]
LGTM

**Review by ririxidev — APPROVED** [review]
lgtm

**Pull request merged**